### DISCUSSION TITLE
Fix to see multiple devices in Alexa

### DISCUSSION THREAD

**Discussion by ForQwer** [vintlabs/fauxmoESP#264] [Show and tell]
In the library itself in fauxmoESP.cpp change the old unsigned char fauxmoESP::addDevice(const char * device_name)

With:

unsigned char fauxmoESP::addDevice(const char * device_name) {

    fauxmoesp_device_t device;
    unsigned int device_id = _devices.size();

    // Initialize properties
    device.name = strdup(device_name);
    device.state = false;
    device.value = 0;

    // Reverting to the original uniqueid format but adding unique suffixes to avoid conflicts
    String mac = WiFi.macAddress();
    int suffix = 10 + device_id;  // Adding 10 to device_id to ensure uniqueness but keep format

    snprintf(device.uniqueid, 27, "%s:%02X:00:00-%02X", mac.c_str(), suffix, device_id);

    // Attach the device
    _devices.push_back(device);

    DEBUG_MSG_FAUXMO("[FAUXMO] Device '%s' added as #%d with unique ID '%s'\n", device_name, device_id, device.uniqueid);

    return device_id;
}

**Comment by nelsonii** [comment]
Thank for your posting this fix. My clients started getting the same error when learning new devices. Existing stuff worked, but new ones did not. (I have one ESP8266 which does control for about 18 different functions -- so this is a big deal.)

If you can get a pull request in, this can be brought into the main code base.

**Comment by conane4lbabaro** [comment]
Thanks for publishing, the correction, but added the line

int suffix = 10 + device_id;  // Adding 10 to device_id to ensure uniqueness but keep format

in my file  fauxmoESP.cpp from my downloaded library

but it has not solved the problem, it still detects only one device, I have to do something else, thank you very much for commenting

**Reply by ForQwer** [reply]
Please try to paste the whole code that I,ve poated

**Reply by conane4lbabaro** [reply]
It works correctly, thank you very much for the help

the library should be modified
